src/main/java/com/amazon/ion/impl/bin/WriteBuffer.java

```diff
@@ -1274,148 +1313,93 @@ public void writeLowerNibbleAt(final long position, final long value) {
         block.data[offset] = (byte) (bitValue & 0xF0 | value) ;
     }
 
-    /** Get the length of FlexInt for the provided value. */
-    public static int flexIntLength(final long value) {
-        int numMagnitudeBitsRequired;
-        if (value < 0) {
-            int numLeadingOnes = Long.numberOfLeadingZeros(~value);
-            numMagnitudeBitsRequired = 64 - numLeadingOnes;
-        } else {
-            int numLeadingZeros = Long.numberOfLeadingZeros(value);
-            numMagnitudeBitsRequired = 64 - numLeadingZeros;
-        }
-        return numMagnitudeBitsRequired / 7 + 1;
-    }
-
     /** Writes a FlexInt to this WriteBuffer, returning the number of bytes that were needed to encode the value */
     public int writeFlexInt(final long value) {
-        int numBytes = flexIntLength(value);
-        return writeFlexIntOrUInt(value, numBytes);
+        int numBytes = FlexInt.flexIntLength(value);
+        writeFlexIntOrUIntAt(position(), value, numBytes);
+        reserve(numBytes);
```

Comment:
That's because if I call `reserve()` first, then `position()` will return the location _after_ the reserved bytes. I'm going to refactor it a bit so that reserve can be called first.